controllers/volumesnapshot.go

```diff
@@ -157,7 +162,7 @@ func (r *VolumeSnapshotBackupReconciler) buildVolumeSnapshotContentClone(vscClon
 		vscClone.Spec = newSpec
 	}
 
-	if vsb.Status.VolumeSnapshotClassName == "" {
+	if vsb.Status.VolumeSnapshotClassName == "" && vscClone != nil && len(*vscClone.Spec.VolumeSnapshotClassName) > 0 {
```

Comment:
remove this, check if the name is empty before this. `vscClone` will never be nil its not necessary